### PR TITLE
Updated Arduino to 1.6.0

### PR DESCRIPTION
**Pull request by dehaas-j**
I updated the arduino expression to build version 1.6.0 rather than 1.0.6. postInstall ensures that the rpaths of the prebuilt binaries (which Arduino's build process pulls in)  are set so that all shared libraries are found. 

The expression also creates a .desktop file if withGui == true.

### CODE REVIEW

**Comment by dajero** [comment]
Removed some leftover debug echo statements from postInstall

File: pkgs/development/arduino/arduino-core/default.nix [hunk 1]
```diff
@@ -1,29 +1,35 @@
-{ stdenv, fetchFromGitHub, jdk, jre, ant, coreutils, gnugrep, file, libusb
-, withGui ? false, gtk2 ? null
+{ stdenv, fetchFromGitHub, jdk, jre, ant, coreutils, gnugrep, file, libusb, unzip,
+  gccRaw, readline, zlib, ncurses, withGui ? false, gtk2 ? null
```
Comment (vcunat) [inline pkgs/development/arduino/arduino-core/default.nix]: I don't see why you have ``gccRaw = gcc.cc``. Why not simply use ``stdenv.cc`` instead?

**Comment by vcunat** [comment]
CC maintainers: @antono, @robberer, @bjornfor.

**Comment by dajero** [comment]
I think partly my unfamiliarity with packaging with Nix was to blame for using `gccRaw = gcc.cc`. I am now using `stdenv.cc.cc` instead (as is done in the `skype` package for instance)

**Comment by bjornfor** [comment]
@dajero: Some suggestions:

- Please rework your PR so that each commit does one logical change.
- Prefix commit message with package name; makes it easier to skim through git history.
- Write commit messages in imperative style (e.g. "update" instead of "updated"). This matches commit messages by git from commands such as git merge and git revert.
- Add descriptive commit messages, so that one doesn't have to refer to github to know the context I'm thinking particularly about the RPATH stuff you mention here... what's currently wrong? Do you have an example of a failure?

**Comment by dajero** [comment]
Alright I will close this pull request and file a new one. I do think that in the end it is just one logical change

**Comment by bjornfor** [comment]
@dajero: You can force push to update existing PR.

**Pull request closed without merging**